### PR TITLE
PLANET-1823 auto set featured image

### PR DESCRIPTION
**Pull request by angtheod**
Automatically set as featured image of a Post the first image found within the Post's content. Allow Editors to override this by setting manually another featured image of their choosing.

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -143,6 +144,53 @@ protected function hooks() {
...
+			update_post_meta( $post_id, 'user_set_featured_image', false );
+		}
+		$user_featured_image = get_post_meta( $post_id, 'user_set_featured_image', true );
+
+		// Apply this behavior to Posts only.
+		if ( 'post' === $post->post_type && ! $user_featured_image ) {
+
+			// Find all matches of <img> html tags within the post's content and get the url inside the src attribute.
+			preg_match_all( '/<img.+src=[\'"]([^\'"]+)[\'"].*>/i', $post->post_content, $matches );
+			$first_img_url = $matches[1][0];
```
Comment (sagarsdeshmukh) [inline functions.php]: If no image found, it will throw a notice level error.

File: functions.php [hunk 2]
```diff
@@ -143,6 +144,53 @@ protected function hooks() {
...
+			// Use the attachment's url to find its id.
+			$statement     = $wpdb->prepare( "SELECT ID FROM $wpdb->posts WHERE guid='%s';", $first_img_url );
+			$result        = $wpdb->get_col( $statement );
+			$attachment_id = $result[0];
```
Comment (sagarsdeshmukh) [inline functions.php]: Same for `$result[0]` as well.

File: functions.php (hunk 1 above)
Comment (angtheod) [reply functions.php]: Yes, saw that, will fix it.

File: assets/admin/js/edit_post.js [hunk 3]
```diff
@@ -49,4 +49,38 @@ $(document).ready(function () {
       }
     }
   });
+
+  // Custom handling of Featured Image selection, so that we know if Editor tried to override the
+  // auto-selected first image found within the Post's content.
+  $("#postimagediv").off("click", "#set-post-thumbnail").on("click", "#set-post-thumbnail", function () {
+  	if ( typeof wp !== "undefined" ) {
```
Comment (sagarsdeshmukh) [inline assets/admin/js/edit_post.js]: For indentation need to use spaces.

File: assets/admin/js/edit_post.js [hunk 4]
```diff
@@ -49,4 +49,38 @@ $(document).ready(function () {
       }
     }
   });
+
+  // Custom handling of Featured Image selection, so that we know if Editor tried to override the
+  // auto-selected first image found within the Post's content.
+  $("#postimagediv").off("click", "#set-post-thumbnail").on("click", "#set-post-thumbnail", function () {
```
Comment (kirdia) [inline assets/admin/js/edit_post.js]: Some eslint warnings about double quotes
  55:5   error  Strings must use singlequote                        quotes
  55:26  error  Strings must use singlequote                        quotes
  55:35  error  Strings must use singlequote                        quotes
  55:61  error  Strings must use singlequote                        quotes
  55:70  error  Strings must use singlequote                        quotes

**Pull request merged**